src/util/amount.rs

```diff
@@ -0,0 +1,868 @@
+            };
+        }
+
+        Ok(Amount::from_sat(value))
+    }
+
+    /// Express this Amount as a floating-point value in the given denomination.
+    ///
+    /// Please be aware of the risk of using floating-point numbers.
+    pub fn to_float_in(&self, denom: Denomination) -> f64 {
+        (self.as_sat() as f64) * 10_f64.powi(denom.precision())
+    }
+
+    /// Express this Amount as a floating-point value in Bitcoin.
+    ///
+    /// Equivalent to `to_float_in(Denomination::Bitcoin)`.
+    ///
+    /// Please be aware of the risk of using floating-point numbers.
+    pub fn as_btc(&self) -> f64 {
+        self.to_float_in(Denomination::Bitcoin)
+    }
+
+    /// Convert an amount in floating-point notation with a given denomination.
+    /// Can return [ParseAmountError::TooBig] or [ParseAmountError::TooPrecise].
+    ///
+    /// Please be aware of the risk of using floating-point numbers.
+    pub fn from_float_in(value: f64, denom: Denomination) -> Result<Amount, ParseAmountError> {
+        // This is inefficient, but the safest way to deal with this. The parsing logic is safe.
+        // Any performance-critical application should not be dealing with floats.
+        Amount::from_str_in(&value.to_string(), denom)
+    }
+
+    /// Format the value of this Amount in the given denomination.
+    ///
+    /// Does not include the denomination.
+    pub fn fmt_value_in(&self, f: &mut fmt::Write, denom: Denomination) -> fmt::Result {
+        if denom.precision() > 0 {
+            // add decimal point and zeroes
+            let width = denom.precision() as usize;
+            write!(f, "{}.{:0width$}", self.as_sat(), 0, width = width)?;
+        } else if denom.precision() < 0 {
+            // need to inject a comma in the number
+            let nb_decimals = denom.precision().abs() as usize;
+            let real = format!("{:0width$}", self.as_sat(), width = nb_decimals);
+            if real.len() == nb_decimals {
+                write!(f, "0.{}", &real[real.len() - nb_decimals..])?;
+            } else {
+                write!(
+                    f,
+                    "{}.{}",
+                    &real[0..(real.len() - nb_decimals)],
+                    &real[real.len() - nb_decimals..]
+                )?;
+            }
+        } else {
+            // denom.precision() == 0
+            write!(f, "{}", self.as_sat())?;
+        }
+        Ok(())
+    }
+
+    /// Get a string number of this Amount in the given denomination.
+    ///
+    /// Does not include the denomination.
+    pub fn to_string_in(&self, denom: Denomination) -> String {
+        let mut buf = String::new();
+        self.fmt_value_in(&mut buf, denom).unwrap();
+        buf
+    }
+
+    /// Get a formatted string of this Amount in the given denomination, suffixed with the
+    /// abbreviation for the denomination.
+    pub fn to_string_with_denomination(&self, denom: Denomination) -> String {
```

Comment:
Isn't this the same thing as `to_string_in` right above it?

Reply:
No, `to_string_in` is just the amount (i.e. `0.42`), this is "with denomination" (i.e. `0.42 BTC`). I think the former has value.

Reply:
Oh, sorry, my bad. Agreed. :)